### ISSUE TITLE
Language Driver request for Elixir

### ISSUE THREAD

**Issue by khari998** [vaticle/typedb-driver#586]
TypeDB is awesome and I definitely want to be able to use it easily in Phoenix/Elixir applications.

**Comment by Sergiusz43** [comment]
An Erlang driver would be wonderful, for Elixir but also for Gleam, which is slowly but surely gaining popularity. Having TypeDB available for this is another big step towards being able to model data for the future. I hope you keep this in mind after the 3.0 release. I thank you very much for your work.

**Comment by farost** [comment]
For future reference: we could use [NIFs](https://www.erlang.org/doc/system/nif.html) for calling our base C driver library, but we'll need to write the wrappers ourselves (manually or by creating an automation, if possible). 

SWIG is not supported. However, there is a crate called [rustler](https://crates.io/crates/rustler) that we could use to write these NIFs easier and faster.

It deserves further investigation.

**Comment by julienmarie** [comment]
Yes creating Rust NIFs with Rustler has become the preferred way thanks to Rust's memory safety.
